pkg/controller/publisher.go

```diff
@@ -71,35 +23,31 @@ func (driver *Controller) ControllerPublishVolume(ctx context.Context, req *csi.
 	if req.GetVolumeCapability() == nil {
 		return nil, status.Error(codes.InvalidArgument, "cannot publish volume without capabilities")
 	}
+
+	nodeIP := req.GetNodeId()
 	parameters := req.GetVolumeContext()
 
-	initiatorMapNodeID := common.GetTopologyCompliantNodeID(req.GetNodeId())
-	var initiatorNames []string
+	var reqType pb.InitiatorType
+	switch parameters[common.StorageProtocolKey] {
+	case common.StorageProtocolSAS:
+		reqType = pb.InitiatorType_SAS
+	case common.StorageProtocolFC:
+		reqType = pb.InitiatorType_FC
+	case common.StorageProtocolISCSI:
+		reqType = pb.InitiatorType_ISCSI
+	}
 
-	// Available initiators for the node are provided in parameters through NodeGetInfo
-	if parameters[common.StorageProtocolKey] == common.StorageProtocolSAS {
-		for key, val := range parameters {
-			if strings.Contains(key, common.TopologySASInitiatorLabel) && strings.Contains(key, initiatorMapNodeID) {
-				initiatorNames = append(initiatorNames, val)
-			}
-		}
-	} else if parameters[common.StorageProtocolKey] == common.StorageProtocolFC {
-		for key, val := range parameters {
-			if strings.Contains(key, common.TopologyFCInitiatorLabel) && strings.Contains(key, initiatorMapNodeID) {
-				initiatorNames = append(initiatorNames, val)
-			}
-		}
-	} else {
-		initiatorNames = []string{req.GetNodeId()}
+	initiators, err := node_service.GetNodeInitiators(nodeIP, reqType)
```

Comment:
must this be an IP address or can it just be a string (e.g. node name)?

Reply:
It could be any kind of address (DNS name or IP), changed to be more generic.